### ISSUE TITLE
Remember indentation

### ISSUE THREAD

**Issue by OscarGodson** [OscarGodson/EpicEditor#39]
This is super useful for writing code. I.e.

If i type (where the pipe is the cursor):

```
if(){
  alert() |
}
```

and press enter it should appear in line with the `alert()` like

```
if(){
  alert()
  |
}
```

**Comment by OscarGodson** [comment]
From @janlimpens in #104

> Hitting enter again, should "outdent" by one level. It might also be good to be able to use the tab key to insert 2 spaces, but this might break UI Accessibility guidelines.

**Comment by fuddl** [comment]
is this also supposed to work on stuff like lists, blockquotes and so on?

**Comment by OscarGodson** [comment]
Yeah. The editor doesn't know what kind of text you're writing. Once you press tab it will keep that indent until you outdent.
